app/scripts/check_requirements

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env php
+<?php
+
+// Script should not be called from the Web
+if (php_sapi_name() != 'cli') {
+    die('Nope');
+}
+
+/**
+ * Colorize the shell output
+ *
+ * @param string $text The text.
+ * @param string $color Color of the text: green, red or nothing.
+ * @return string
+ */
+function colorizeOutput($text, $color) {
+    switch ($color) {
+            case 'green':
+                // White on green
+                $output = "\033[1;37m\033[42m{$text}";
+                break;
+            case 'red':
+                // White on red
+                $output = "\033[1;37m\033[41m{$text}";
+                break;
+            default:
+                $output = $text;
+                break;
+        }
+        $output .= "\033[0m";
+        return $output;
+    }
+
+/**
+ * Check current configuration.
+ * 
+ * @param bool   $condition The type of alert.
+ * @param string $message   The message to display.
+ * @param string $help      Help about the message.
+ * @param string $suggested_fix A message that help to fix the issue.
+ *
+ * @return void
+ */
+function check($condition, $message, $help = '', $suggested_fix = '') {
+    $result = $condition ? colorizeOutput('OK', 'green') : colorizeOutput('FAILED', 'red');
+    $text = "* {$message}: {$result}\n";
+
+    if (! $condition) {
+        if(! empty($help)) { 
+            $text .=  "  {$help}\n";
+        }
+        if(! empty($suggested_fix)) {
+            $text .=  "  {$suggested_fix}\n";
+        }
+    }
+    
+    echo $text;
+}
+
+/**
+ * Check if the directory is empty
+ *
+ * @param string $dir The path.
+ * @return bool
+ */
+function isEmptyDir($dir) {
+    if(!file_exists($dir) || !is_dir($dir)) {
+        return false;
+    }
+    if (($files = @scandir($dir)) && count($files) <= 2) {
+        return true;
+    }
+        
+    return false;
+}
+
+echo " Transvision Requirements Check\n";
+
+// Get the parent folder from the script itself
+$root = dirname(dirname(dirname(__FILE__)));
+
+//Load settings and constants
+$server_config = parse_ini_file($root . '/app/config/config.ini');
+include_once($root . '/app/inc/constants.php');
+
+// Check PHP version
+$php_version = phpversion();
+check(version_compare($php_version, '5.4', '>='), sprintf('PHP version is at least 5.4 (%s)', $php_version), 'Current version is ' . $php_version);
+
+// Check permissions on cache folder for the user
+clearstatcache(null, CACHE_PATH);
+$cachePerms = substr(sprintf('%o', fileperms(CACHE_PATH) ), -4);
+$command = 'chmod -R 755 ' . CACHE_PATH;
+check($cachePerms[1] == '7', 'Cache folder is writable by the user running this script', sprintf('Cache folder should be writable by the user running the script (current permissions : (%s))', $cachePerms), "To fix execute: " . $command);
```

Comment:
For success: "Cache folder is writable by the user running this script. Make sure it's also writable by the user running PHP on the webserver".

For failure: "Cache folder should be writable (current permissions: {$cachePerms}). This warning can be ignored as long as the user running PHP on the webserver can write in the folder."

Again, drop the sprintf and add the variable to the string where possible.